packages/commonwealth/client/scripts/views/pages/Communities/TokenLaunchDrawer/QuickTokenLaunchForm/QuickTokenLaunchForm.tsx

```diff
@@ -0,0 +1,302 @@
+import { ChainBase, commonProtocol } from '@hicommonwealth/shared';
+import clsx from 'clsx';
+import { notifyError } from 'controllers/app/notifications';
+import useBeforeUnload from 'hooks/useBeforeUnload';
+import React, { useState } from 'react';
+import { slugifyPreserveDashes } from 'shared/utils';
+import { useUpdateCommunityMutation } from 'state/api/communities';
+import useCreateCommunityMutation, {
+  buildCreateCommunityInput,
+} from 'state/api/communities/createCommunity';
+import { generateImage } from 'state/api/general/generateImage';
+import { useLaunchTokenMutation } from 'state/api/launchPad';
+import { useCreateTokenMutation } from 'state/api/token';
+import useUserStore from 'state/ui/user';
+import PageCounter from 'views/components/PageCounter';
+import { CWText } from 'views/components/component_kit/cw_text';
+import CWBanner from 'views/components/component_kit/new_designs/CWBanner';
+import { CWButton } from 'views/components/component_kit/new_designs/CWButton';
+import CWCircleMultiplySpinner from 'views/components/component_kit/new_designs/CWCircleMultiplySpinner';
+import TokenLaunchButton from 'views/components/sidebar/TokenLaunchButton';
+import { generateCommunityNameFromToken } from '../../../LaunchToken/steps/CommunityInformationStep/utils';
+import SuccessStep from '../../../LaunchToken/steps/SuccessStep';
+import TokenInformationForm from '../../../LaunchToken/steps/TokenInformationStep/TokenInformationForm';
+import { FormSubmitValues } from '../../../LaunchToken/steps/TokenInformationStep/TokenInformationForm/types';
+import useCreateTokenCommunity from '../../../LaunchToken/useCreateTokenCommunity';
+import './QuickTokenLaunchForm.scss';
+
+type QuickTokenLaunchFormProps = {
+  onCancel: () => void;
+  onCommunityCreated: (communityId: string) => void;
+};
+
+export const QuickTokenLaunchForm = ({
+  onCancel,
+  onCommunityCreated,
+}: QuickTokenLaunchFormProps) => {
+  const [randomizeAttempts] = useState<{ data: Object }[]>([]);
+  const [isCreatingQuickToken, setIsCreatingQuickToken] = useState(false);
+  const [
+    createdCommunityIdsToTokenInfoMap,
+    setCreatedCommunityIdsToTokenInfoMap,
+  ] = useState({});
+
+  const {
+    selectedAddress,
+    setSelectedAddress,
+    baseNode,
+    createdCommunityId,
+    setCreatedCommunityId,
+  } = useCreateTokenCommunity();
+
+  const { mutateAsync: launchToken } = useLaunchTokenMutation();
+
+  const { mutateAsync: createToken } = useCreateTokenMutation();
+
+  const { mutateAsync: updateCommunity } = useUpdateCommunityMutation({
+    communityId: createdCommunityId || '', // this will exist by the time update community is called
+  });
+
+  const user = useUserStore();
+
+  const { mutateAsync: createCommunityMutation } = useCreateCommunityMutation();
+
+  useBeforeUnload(isCreatingQuickToken);
+
+  const handleSubmit = (tokenInfo: FormSubmitValues) => {
```

Comment:
This handler is sick. There are multiple things here that could go wrong. Do we handle properly failure of each step? What if the failure is somewhere in the middle of the flow? Do we rollback previous steps or this is fine to keep them in this state?

Reply:
There are 4 cases where the handler can break
1. during image generation via AI => `await generateImage()`
2. during community generation => `await createCommunityMutation`
3. during token creation/launch => `await launchToken() and await createToken()`
4. during update community after every other API call is complete => `await updateCommunity()`

the first 3 are not an issue, failure cases are handled for them. As for 4, I will add a failure case.

For a longer explanation if you are interested (apologies for the wall of text below 😅)

1. When this happens, the handler will fail without side effects since we wouldn't have mutated any data on the API level.
2. I already handled cases where this call can fail due to a non-unique community ID/name. If it fails for some other reason, the handler will fail without side effects, since we wouldn't have created/mutated anything on the API side yet.
3. If this step fails 
    a. at `await launchToken()`, then it means step 2 succeeded, which means we created a community, now if the user submits the form again, we check if a community for the same "submitted" token info was created via the `createdCommunityIdsToTokenInfoMap` local state, and if it was we skip step `2` (community creation) and jump straight to step 3. This means failure of the `await launchToken()` call won't have any side effects, unless the user forcibly closes the browser tab, and the community created will remain - we could probably delete this community (needs super admin privilege) but I think it is fine for now. 
    b. at `await createToken()`, the only viable reason (since we reached this far without any chain node/chain base issues) would be that the token name already exists. This can happen in a small (20-30 sec) window where a different user (User - B) would have to enter the exact info from another device, sign the token transaction first and take the token name, and the first user (User - A) would have to delay the exact same time (20-30 sec) for token transaction signing (a highly rare case). Ideally, to solve these types of cases we can use "action_intents" on the API side, where we store the draft state of any "records" on the API side (these draft records can have duplicates/non-unique properties even if their production entities don't allow that). Once the user has actually done the final "confirmation" for that drafted "action_intent", it gets converted to real entity data (if they satisfy non-duplicate/unique aspects of the same entity in production). TBH these types of issues are unaddressed in many flows, like stake/namespace or any form of web3 transactions where we reserve something on the chain. I agree we should handle these rare cases too. Lmk what you think.
4. This is just something complementary I did to keep token/community info in sync (in case the user updates the token description/image after the handler fails once). The failure of this call can be suppressed as it's not really an issue, will add a `.catch()` to this.

Reply:
thank you for detailed explanation. It looks like you thought it through which is great! 

Re 3. I agree that we dont support this kind of error handling in many cases and I am not saying that you should do it here. Also you described how this failure would happen on the user side. Probability is very low what you described. I was thinking more about server side issue like db/be errors bc of downtime or wrong params from FE etc. So it looks all good to me!